datafusion/core/src/datasource/file_format/csv.rs

```diff
@@ -716,6 +733,10 @@ mod tests {
     async fn query_compress_data(
         file_compression_type: FileCompressionType,
     ) -> Result<()> {
+        let runtime = Arc::new(RuntimeEnv::new(RuntimeConfig::new()).unwrap());
+        let cfg = SessionConfig::new().set_str("datafusion.catalog.has_header", "true");
+        let session_state = SessionState::new_with_config_rt(cfg, runtime);
```

Comment:
I think you can write this like this for extra compile time checking (maybe for some other PR)

```rust
        let mut cfg = SessionConfig::new();
        cfg.options_mut().catalog.has_header = true;
        let session_state = SessionState::new_with_config_rt(cfg, runtime);
```